lib/guardian.rb

```diff
@@ -323,6 +323,12 @@ def can_use_primary_group?(user, group_id = nil)
     (group ? !group.automatic : false)
   end
 
+  def can_use_flair_group?(user, group_id = nil)
+    return false if !user || !group_id || !user.group_ids.include?(group_id.to_i)
+    flair_icon, flair_upload_id = Group.where(id: group_id.to_i).pluck_first(:flair_icon, :flair_upload_id)
+    flair_icon.present? || flair_upload_id.present?
```

Comment:
Sorry I know I said to pluck the columns previously but I realized the logic can be done in SQL land. 

`Group.where(id: group_id.to_i).where("flair_icon IS NOT NULL OR flair_upload_id IS NOT NULL").exists?`

Reply:
If you do not mind, I would like to keep this as it is as I find it easier to comprehend. There is going to be a similar query either way ('SELECT 1' or 'SELECT flair_icon, flair_upload_id'), so this looks like a micro-optimization to me.

Moreover, your suggestion is not equivalent to the initial code. `IS NOT NULL` returns true for empty string, but `present?` returns false.

Reply:
I don’t think those columns will ever have a blank string as its value but I understand your concern 👍

Reply:
They will have a blank string everytime an admin adds and then removes the flair.